src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmDirectValueAnnotationsManager.cs

```diff
@@ -245,8 +249,14 @@ private static IEdmDirectValueAnnotation FindTransientAnnotation(object transien
                 else
                 {
                     VersioningList<IEdmDirectValueAnnotation> annotationsList = (VersioningList<IEdmDirectValueAnnotation>)transientAnnotations;
-                    return annotationsList.FirstOrDefault(
-                        existingAnnotation => existingAnnotation.NamespaceUri == namespaceName && existingAnnotation.Name == localName);
+

```

Comment:
It hurts me a bit to see that it makes such a change and sacrificing readability. But it is worth the savings.
Can we please add a comment about the foreach stating that for the purpose of optimization, this is not written using Enumerable.Any since this allocates a closure every time.

Reply:
I don't think adding a comment is really warranted IMHO, otherwise we would need to add such comments everywhere? :) I mean, it boils down to simple coding best practices that should be followed in the code base, like avoiding Linq in the hot path. I thought many of these best practices were already being adopted, as I have reviewed multiple PRs in the past that were specifically to fix that in OData? Granted, ideally they should be written down somewhere, not sure that is the case.